### DISCUSSION TITLE
Better developer efficiency

### DISCUSSION THREAD

**Discussion by iliya-fatahi** [community/community#46393] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I feel that I am not productive enough for the company I work in and this is a bad feeling. I would appreciate it if you could give me some solutions to measure my productivity or how to improve it or...

**Comment by canaleal** [comment, marked as answer]
Hey @iliya-fatahi, Feeling unproductive can be frustrating, but it's important to remember that everyone experiences dips in productivity from time to time and that productivity as a whole is relative and hard to measure. 

Either way, remember that productivity is not just about working harder or faster, but about working smarter. By setting clear goals, prioritizing your tasks, and minimizing distractions, you can make the most of your time and achieve your goals more efficiently.

### Possible Solution

While I can't give you an full proof way to measure your progress, here are some tools that I personally used to measure how productive I've been throughout the week. 

- [Wakatime](https://wakatime.com/) to measure how much time I've spent writing, reviewing, and managing code.
- I write 3 Key Performance Area tasks (KPA) at the start of every week and then see how many I can complete by the end of the month. One example is:
`Maintain at least 80% code test coverage throughout Feb`
- [Todoist](https://todoist.com/) to track any none JIRA tasks that I need to complete throughout the day. 
- I carry a log book of all the tasks I completed that day and how long each task took. I can then use this data to measure if I'm spending too much time on a given task. 

I would also recommend talking to your team and/or manager and seeing if you're on track with your work or if you need to step up and do more.

**Reply by iliya-fatahi** [reply]
Thanks for say your tips
